frontend/src/lib/components/LemonButton/LemonButton.tsx

```diff
@@ -153,7 +153,7 @@ export function LemonButtonWithPopup({
         if (!buttonProps.icon) {
             buttonProps.icon = popupProps.placement?.startsWith('right') ? <IconChevronRight /> : <IconArrowDropDown />
         }
-    } else if (!buttonProps.sideIcon) {
+    } else if (buttonProps.sideIcon === undefined) {
```

Comment:
I think this still aligns with expected behavior, since passing in an explicit null here would mean that no side icon would be rendered instead of the default chevron. Ideally there would be another prop called `shouldShowSideIcon` or something like that which controls rendering the icon at all, but I'm not sure if that's worth the overhead.